doc/plugins.dox

```diff
@@ -358,81 +464,79 @@ Once this is done, your first plugin is ready. Congratulations!
 @section listing Complete listing (without comments)
 
 @code
-#include <geanyplugin.h>
-
-GeanyPlugin		*geany_plugin;
-GeanyData		*geany_data;
-
-PLUGIN_VERSION_CHECK(211)
 
-PLUGIN_SET_INFO("HelloWorld", "Just another tool to say hello world",
-				"1.0", "John Doe <john.doe@example.org>");
-
-
-static GtkWidget *main_menu_item = NULL;
+#include <geanyplugin.h>
 
-static void item_activate_cb(GtkMenuItem *menuitem, gpointer gdata)
+static void item_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
 {
 	dialogs_show_msgbox(GTK_MESSAGE_INFO, "Hello World");
 }
 
-void plugin_init(GeanyData *data)
+
+static gboolean hello_init(GeanyPlugin *plugin, gpointer pdata)
 {
 	main_menu_item = gtk_menu_item_new_with_mnemonic("Hello World");
 	gtk_widget_show(main_menu_item);
 	gtk_container_add(GTK_CONTAINER(geany->main_widgets->tools_menu),
-		main_menu_item);
+			main_menu_item);
 	g_signal_connect(main_menu_item, "activate",
-		G_CALLBACK(item_activate_cb), NULL);
+			G_CALLBACK(item_activate_cb), NULL);
+
+	geany_plugin_set_data(plugin, main_menu_item, NULL);
+	retrun TRUE;
```

Comment:
maybe we should try to just c&p the example and see if it works to get rid of such possible typos, the compiler is way better at it than we are :)